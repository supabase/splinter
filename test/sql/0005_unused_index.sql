begin;
    set local search_path = '';

  -- No issues
  select * from lint."0005_unused_index";

  create table public.foo (
    id int primary key,
    foo text unique,
    bar text
  );

  create index some_unused_index on public.foo (bar);

  -- Only the "bar" table is listed
  select * from lint."0005_unused_index";

rollback;
