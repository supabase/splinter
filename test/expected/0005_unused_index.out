begin;
  -- No issues
  select * from lint."0005_unused_index";
 name | level | facing | categories | description | detail | remediation | metadata | cache_key 
------+-------+--------+------------+-------------+--------+-------------+----------+-----------
(0 rows)

  create table public.foo (
    id int primary key,
    foo text unique,
    bar text
  );
  create index some_unused_index on public.foo (bar);
  -- Only the "bar" table is listed
  select * from lint."0005_unused_index";
     name     | level |  facing  |  categories   |                                 description                                 |                                detail                                 |                                   remediation                                    |                       metadata                       |                 cache_key                 
--------------+-------+----------+---------------+-----------------------------------------------------------------------------+-----------------------------------------------------------------------+----------------------------------------------------------------------------------+------------------------------------------------------+-------------------------------------------
 unused_index | INFO  | EXTERNAL | {PERFORMANCE} | Detects if an index has never been used and may be a candidate for removal. | Index \`some_unused_index\` on table \`public.foo\` has not been used | https://supabase.com/docs/guides/database/database-linter?lint=0005_unused_index | {"name": "foo", "type": "table", "schema": "public"} | unused_index_public_foo_some_unused_index
(1 row)

rollback;
